File: models/wish.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Category string

const (
	Career    		Category = "진로"
	Health   		  Category = "건강"
	Relationships Category = "인간관계"
	Money     		Category = "돈"
	Goals      		Category = "목표"
	Academics  		Category = "학업/성적"
	Other      		Category = "기타"
)

type WishStatus string

const (
	Approved WishStatus = "approved"
	Pending  WishStatus = "pending"
	Rejected WishStatus = "rejected"
)

type Wish struct {
	ID        int            `json:"id" gorm:"primaryKey"`
	Title     string         `json:"title" validate:"required"`
	Content   string         `json:"content" validate:"required"`
	Category  Category       `json:"category" validate:"required"`
	CreatedAt time.Time      `json:"created_at" gorm:"autoCreateTime"`
	IsConfirm WishStatus     `json:"is_confirm" gorm:"default:'pending'"`
	DeletedAt gorm.DeletedAt `json:"deleted_at" gorm:"index"`
}

type WishInput struct {
	Title    string `json:"title" binding:"required"`
	Content  string `json:"content" binding:"required"`
	Category string `json:"category" binding:"required"`
}

File: src/main/java/goodnight/tree/domain/dao/Wish.java
package goodnight.tree.domain.dao;

import goodnight.tree.global.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@Entity @Getter
// 소프트 삭제 구현 delete 메소드 사용 시 실제 삭제가 아닌 소프트 삭제로 구동
//@SQLDelete(sql = "UPDATE wish SET deleted_at = NOW() WHERE id = ?")
// 매 조회 시 deleted at이 null인 값만
//@Where(clause = "deleted_at IS NULL")
public class Wish extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String content;

    @Enumerated(EnumType.STRING)
    private Category category;

    @Enumerated(EnumType.STRING)
    private WishStatus wishStatus = WishStatus.PENDING;

    @OneToMany(mappedBy = "wish", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments = new ArrayList<>();


    // Getters and Setters, Constructors

    public enum Category {
        CAREER, HEALTH, RELATIONSHIPS, MONEY, GOALS, STUDIES, OTHER
    }

    public enum WishStatus {
        APPROVED, PENDING, REJECTED
    }

    public Wish(String title, String content, Category category) {
        this.title = title;
        this.content = content;
        this.category = category;
    }

    public void changeStatus(WishStatus newStatus) {
        this.wishStatus = newStatus;
    }
}